### PR TITLE
DAOS-5447 build: Update HDF5 1.12 build to SLES15.2

### PR DESCRIPTION
**Pull request by mjean308**
Signed-off-by: Maureen Jean <maureen.jean@intel.com>

### CODE REVIEW

**Comment by mjean308** [comment]
This build is using bmurrell/build-leap-on-15.2-and-ubuntu-20.04 and can not land until packaging files have been updated.
Would like to get feedback on hdf5.spec changes

File: hdf5.spec [hunk 1]
```diff
@@ -249,8 +257,8 @@ ln -s %{_javadir}/slf4j/api.jar java/lib/slf4j-api-1.7.25.jar
 ln -s %{_javadir}/slf4j/nop.jar java/lib/ext/slf4j-nop-1.7.25.jar
 ln -s %{_javadir}/slf4j/simple.jar java/lib/ext/slf4j-simple-1.7.25.jar
 # Fix test output
-junit_ver=$(sed -n '/<version>/{s/^.*>\([0-9]\.[0-9]*\)<.*/\1/;p;q}' /usr/share/maven-poms/JPP-junit.pom)
-sed -i -e "s/JUnit version .*/JUnit version $junit_ver/" java/test/testfiles/JUnit-*.txt
+# junit_ver=$(sed -n '/<version>/{s/^.*>\([0-9]\.[0-9]*\)<.*/\1/;p;q}' /usr/share/maven-poms/JPP-junit.pom)
+# sed -i -e "s/JUnit version .*/JUnit version $junit_ver/" java/test/testfiles/JUnit-*.txt
```
Comment (brianjmurrell) [inline hdf5.spec]: Why are these commented out?  Something you intended to go back and investigate?

File: hdf5.spec [hunk 2]
```diff
@@ -97,6 +97,14 @@ Provides:       %{name}-daos-%{daos_major} = %{version}-%{release}
 %global mpi_list %{?mpi_list} openmpi3
 %endif
 
+%if (0%{?suse_version} >= 1500)
+%global mpi_libdir %{_libdir}/mpi/gcc
+%global mpi_incldir  %{_includedir}/mpi/gcc
+%else
+%global mpi_libdir %{_libdir}
+%global mpi_incldir  %{_includedir}
+%endif
```
Comment (brianjmurrell) [inline hdf5.spec:106]: Why is this necessary just to bump the Leap 15 build version to 15.2?  This seems like something that a version bump should not cause.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: hdf5.spec (hunk 2 above)
Comment (mjean308) [reply hdf5.spec:106]: It is not the version bump but SLES15.      These files should be in the mpi path and for SLES15.* this is /usr/lib64/mpi/gcc/<mpi>.

File: hdf5.spec (hunk 1 above)
Comment (mjean308) [reply hdf5.spec]: Yes it fails with 15.2.    The failure had to do with /usr/share/maven-poms/ not exisiting ad Ineed to figure out why..

Comment (mjean308) [reply hdf5.spec]: filename changed in latest junit rpm to junit.pom; added conditional

File: hdf5.spec [hunk 3]
```diff
@@ -248,8 +257,13 @@ ln -s %{_javadir}/junit.jar java/lib/junit.jar
 ln -s %{_javadir}/slf4j/api.jar java/lib/slf4j-api-1.7.25.jar
 ln -s %{_javadir}/slf4j/nop.jar java/lib/ext/slf4j-nop-1.7.25.jar
 ln -s %{_javadir}/slf4j/simple.jar java/lib/ext/slf4j-simple-1.7.25.jar
+
 # Fix test output
+%if (0%{?suse_version} >= 1500)
+junit_ver=$(sed -n '/<version>/{s/^.*>\([0-9]\.[0-9]*\)<.*/\1/;p;q}' /usr/share/maven-poms/junit.pom)
+%else
 junit_ver=$(sed -n '/<version>/{s/^.*>\([0-9]\.[0-9]*\)<.*/\1/;p;q}' /usr/share/maven-poms/JPP-junit.pom)
+%endif
```
Comment (brianjmurrell) [inline hdf5.spec:266]: For future reference it's usually best to make the difference needed in branches such as this one as small as possible to mitigate future copy-and-paste update errors.  I.e.

```suggestion
%if (0%{?suse_version} >= 1500)
junit_ver_file=junit
%else
junit_ver_file=JPP-junit
%endif
junit_ver=$(sed -n '/<version>/{s/^.*>\([0-9]\.[0-9]*\)<.*/\1/;p;q}' /usr/share/maven-poms/$junit_ver_file.pom)

```

**Review by brianjmurrell — APPROVED** [review]
